src/Representation/Module.cs

```diff
@@ -1,50 +1,65 @@
+        private readonly (string, IStructType?) _vaListType;
 
         public Module(
             string target,
             Bits pointerSize,
-            IStructType? directoryStreamType,
-            IStructType? directoryEntryType,
-            IStructType? jumpBufferType,
-            IStructType? localeType,
-            IStructType? statType,
-            IStructType? threadType,
-            IStructType? vaListType,
+            (string, IStructType?) directoryStreamType,
+            (string, IStructType?) directoryEntryType,
+            (string, IStructType?) jumpBufferType,
+            (string, IStructType?) localeType,
+            (string, IStructType?) statType,
+            (string, IStructType?) threadType,
+            (string, IStructType?) vaListType,
             IFunction[] functions,
             IGlobal[] globals)
         {
             Target = target;
             PointerSize = pointerSize;
-            DirectoryStreamType = directoryStreamType;
-            DirectoryEntryType = directoryEntryType;
-            JumpBufferType = jumpBufferType;
-            LocaleType = localeType;
-            StatType = statType;
-            ThreadType = threadType;
-            VaListType = vaListType;
+            _directoryStreamType = directoryStreamType;
+            _directoryEntryType = directoryEntryType;
+            _jumpBufferType = jumpBufferType;
+            _localeType = localeType;
+            _statType = statType;
+            _threadType = threadType;
+            _vaListType = vaListType;
             _functions = functions;
             _globals = globals;
         }
 
         public string Target { get; }
         public Bits PointerSize { get; }
-        public IStructType? DirectoryStreamType { get; }
-        public IStructType? DirectoryEntryType { get; }
-        public IStructType? JumpBufferType { get; }
-        public IStructType? LocaleType { get; }
-        public IStructType? StatType { get; }
-        public IStructType? ThreadType { get; }
-        public IStructType? VaListType { get; }
+        public IStructType DirectoryStreamType => GetStructType(_directoryStreamType);
+        public IStructType DirectoryEntryType => GetStructType(_directoryEntryType);
+        public IStructType JumpBufferType => GetStructType(_jumpBufferType);
+        public IStructType LocaleType => GetStructType(_localeType);
+        public IStructType StatType => GetStructType(_statType);
+        public IStructType ThreadType => GetStructType(_threadType);
+        public IStructType VaListType => GetStructType(_vaListType);
         public IEnumerable<IFunction> Functions => _functions;
         public IEnumerable<IGlobal> Globals => _globals;
+
+        private static IStructType GetStructType((string, IStructType?) namedStructType)
+        {
+            var (name, structType) = namedStructType;
+
+            return structType ?? throw new Exception($"Struct type {name} was not found.");
```

Comment:
While switching this exception over to the new design we discussed, I noticed that I wasn't actually properly handling the cases where the struct type is not found. So I want to get this in first and I will rebase all the other changes from here. Also note that the larger structural change here is to put the actual type name in the exception message, which is more useful for debugging.